src/LandIce/evaluators/LandIce_EnthalpyResid_Def.hpp

```diff
@@ -26,6 +27,13 @@ Type distance (const Type& x0, const Type& x1, const Type& x2,
     return 0;
 }
 
+template<typename Type>
+KOKKOS_INLINE_FUNCTION
+Type deviceMax(Type a, Type b)
+{
+  return (a > b) ? a : b;
+}
+
```

Comment:
We'll have to make a utility namespace for this sort of thing. We should probably be using the Sacado max too. I'll look into it, maybe we already have something.

Reply:
I think `std::max` should work. I'm pretty sure Sacado overloads it.

Reply:
They were originally std::max and std::min respectively but that resulted in compiler warnings about host code being inaccessible from device and then segmentation faults at runtime.

Reply:
I don't think std::max will work on device which I think means we have to create our own overload.

Reply:
So I think max() is available but it's not in std namespace.

Reply:
It looks like max() and min() are available, the issue was trying to use std::max and std::min. The deviceMax and deviceMin functions are now removed.

Reply:
That's great and good to know. I wonder who's dropping the sacado device max() into the global namespace.

Reply:
I built this on Blake to make sure everything works for serial and the compiler couldn't find max() and min(). We'll need to find a portable solution for this after all.

Reply:
Okay I think the best option will be to create some sort of utility namespace for this. We can discuss later today.

Reply:
This seems very odd. Have you tried to include cmath? Perhaps intel's header are different from gcc (or other intel's other than blake, for some reason); cmath might have been included indirectly in other builds, but not here.

Another solution could be this:

```
#ifndef KOKKOS_ENABLE_CUDA
template <typename T>
KOKKOS_INLINE_FUNCTION
const T& max (const T& a, const T& b) { return a > b ? a : b; }
#else
using std::max;
#endif
```

Reply:
Your latter suggestion is what I was thinking but I think we should put it somewhere for general use (along with min).

Reply:
Oh, yeah, it shouldn't be buried deep inside some evaluator class Def file. Maybe `Albany_Utils.hpp`, but that file may become a panacea. Perhaps `Albany_KokkosUtils.hpp` or `Albany_MathUtils.hpp`.

Reply:
`Albany_KokkosUtils.hpp` sounds good.